ntoskrnl/ex/keyedevt.c

```diff
@@ -457,8 +457,11 @@ NtWaitForKeyedEvent(
     /* Do the wait */
     Status = ExpWaitForKeyedEvent(KeyedEvent, Key, Alertable, Timeout);
 
-    /* Dereference the keyed event */
-    ObDereferenceObject(KeyedEvent);
+    if (Handle != NULL)
```

Comment:
Practically speaking, it's impossible to get here in the `Handle == NULL` case because we would have already bugchecked in ExpWaitForKeyedEvent, since ExpCritSecOutOfMemoryEvent is always NULL.